lib/protocol/protocol.js

```diff
@@ -35,31 +36,61 @@ const IMG_EL_URL_RE = new RegExp(
 
 class Protocol {}
 
-function getLogByProtocol (protocol) {
-  switch (protocol) {
-    case BaseDriver.DRIVER_PROTOCOL.W3C:
-      return w3cLog;
-    case BaseDriver.DRIVER_PROTOCOL.MJSONWP:
-      return mjsonwpLog;
-    default:
-      return genericProtocolLog;
+class SessionsCache {
+  constructor (max) {
+    this._cache = new LRU({ max });
+  }
+
+  getLogger (sessionId, protocol) {
+    if (sessionId) {
+      if (this._cache.has(sessionId)) {
+        const value = this._cache.get(sessionId);
+        if (value.logger) {
+          return value.logger;
+        }
+        protocol = protocol || value.protocol;
+      }
+      return logger.getLogger(`${protocol || GENERIC_PROTOCOL} ` +
```

Comment:
Should this get added to `value`? If, for whatever reason, there is no logger for a session id, each log line will create a new logger, which is then thrown out.

Reply:
This was done intentionally to include the "wrong" session identifier into the actual log lines without caching the whole logger. 

Although, if we don't want to create a new instance each time this method is called, then we could either fall back to the default logger with protocol name only or include some generic prefix there, like `[W3C (unknown session)]`.

Reply:
We don't want to put each random session id into the cache in order to save memory. Instead we only put these ids that are returned by `createSession` call

Reply:
Ok. This should be documented, or else it will be "fixed" at some later date.

Reply:
I'll add a comment